src/argilla/client/datasets.py

```diff
@@ -360,16 +368,172 @@ def _from_pandas(cls, dataframe: pd.DataFrame) -> "Dataset":
+            ...     )
+            ... ])
+            >>> rb_dataset.prepare_for_training().features
+            {'text': Value(dtype='string'),
+             'tokens': Sequence(feature=Value(dtype='string'), length=-1),
+             'prediction': Value(dtype='null'),
+             'prediction_agent': Value(dtype='null'),
+             'annotation': [{'end': Value(dtype='int64'),
+               'label': Value(dtype='string'),
+               'start': Value(dtype='int64')}],
+             'annotation_agent': Value(dtype='null'),
+             'id': Value(dtype='null'),
+             'metadata': Value(dtype='null'),
+             'status': Value(dtype='string'),
+             'event_timestamp': Value(dtype='null'),
+             'metrics': Value(dtype='null'),
+             'ner_tags': [ClassLabel(num_classes=3, names=['O', 'B-TAG', 'I-TAG'])]}
+
+            >>> import argilla as rg
+            >>> rb_dataset = rg.DatasetForTextClassification([
+            ...     rg.TextClassificationRecord(
+            ...         inputs={"header": "my header", "content": "my content"},
+            ...         annotation="SPAM",
+            ...     )
+            ... ])
+            >>> rb_dataset.prepare_for_training().features
+            {'header': Value(dtype='string'),
+             'content': Value(dtype='string'),
+             'label': ClassLabel(num_classes=1, names=['SPAM'])}
+
+        """
+        # determine split fragments
+        if train_size is None and test_size is None:
+            pass
+        elif train_size is None:
+            assert 0 < test_size < 1, ValueError("test_size must be between 0 and 1.")
+            train_size = 1 - test_size
+        elif test_size is None:
+            assert 0 < train_size < 1, ValueError("train_size must be between 0 and 1.")
+            test_size = 1 - train_size
+        else:
+            assert sum([train_size, test_size]) == 1, ValueError(
+                "train_size and test_size must sum to 1."
+            )
+
+        # check for annotations
+        assert any([rec.annotation for rec in self._records]), ValueError(
+            "Dataset has no annotations."
+        )
+
+        # turn the string into a Framework instance and trigger error if str is not valid
+        if isinstance(framework, str):
+            framework = Framework(framework)
+
+        # prepare for training for the right method
+        if framework is Framework.TRANSFORMERS:
+            return self._prepare_for_training_with_transformers(
+                train_size=train_size, test_size=test_size
+            )
+        elif framework is Framework.SPACY and lang is None:
+            raise ValueError(
+                "Please provide a spacy language model to prepare the"
+                " dataset for training with the spacy framework."
+            )
+        elif framework is Framework.SPACY or framework is Framework.SPARK_NLP:
+            if train_size or test_size:
```

Comment:
Could we assume a default value of `1.0` for `train_size` and always use the `train_test_split` function? By default, this function applies a `shuffle` that may be cool to apply for preparing the training data. What do you think?